app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-page-views.js

```diff
@@ -92,6 +93,10 @@ window.GOVUK.analyticsGa4.analyticsModules = window.GOVUK.analyticsGa4.analytics
       }
     },
 
+    getElementAttribute: function (attributeName) {
+      return document.querySelector('[' + attributeName + ']').getAttribute(attributeName)
```

Comment:
I wonder if moving more of the logic above into this function would make this a little clearer, something like this:

```JavaScript
phase_banner: this.getElementAttribute('data-ga4-phase-banner') || undefined
```

```JavaScript
getElementAttribute: function (attributeName) {
  var el = document.querySelector('[' + attributeName + ']')
  if (el) {
    return el.getAttribute(attributeName)
  }
},
```

Reply:
@andysellick Good idea - I've added this, thanks.